pennylane_lightning/core/src/simulators/lightning_kokkos/gates/GateFunctorsGenerator.hpp

```diff
@@ -972,4 +972,35 @@ template <class PrecisionT, bool adj = false> struct generatorMultiRZFunctor {
             1 - 2 * int(Kokkos::Impl::bit_count(k & wires_parity) % 2));
     }
 };
+
+template <class PrecisionT> struct collapseFunctor {
```

Comment:
Generators are gate derivatives (up to a factor). This should rather go in the gate functor module. As a rule of thumb, if the constructor does not do any preprocessing a KOKKOS_LAMBDA should be used instead. I would therefore move the implementation to the SV module instead of the gate module.